pimd/pim_sock.c

```diff
@@ -313,12 +420,90 @@ int pim_socket_join(int fd, struct in_addr group, struct in_addr ifaddr,
 	}
 
 	return ret;
+
+}
+
+static inline int pim_socket_join_v6(int fd, pim_addr group, pim_addr ifaddr,
+			      ifindex_t ifindex)
+{
+	int ret;
+
+	struct ipv6_mreq opt;
+
+	memcpy(&opt.ipv6mr_multiaddr,&group,16);
+
+	opt.ipv6mr_interface = ifindex;
+
+	ret = setsockopt(fd, IPPROTO_IPV6, IPV6_JOIN_GROUP, &opt, sizeof(opt));
+	if (ret) {
+		char group_str[INET6_ADDRSTRLEN];
+		char ifaddr_str[INET6_ADDRSTRLEN];
+		if (!inet_ntop(AF_INET6, &group, group_str, sizeof(group_str)))
+			snprintf(group_str, sizeof(group_str), "<group?>");
+		if (!inet_ntop(AF_INET6, &ifaddr, ifaddr_str,
+			       sizeof(ifaddr_str)))
+			snprintf(ifaddr_str, sizeof(ifaddr_str), "<ifaddr?>");
+
+		flog_err(
+			EC_LIB_SOCKET,
+			"Failure socket joining fd=%d group %s on interface address %s: errno=%d: %s",
+			fd, group_str, ifaddr_str, errno, safe_strerror(errno));
+		return ret;
+	}
+
+	if (PIM_DEBUG_TRACE) {
+		char group_str[INET6_ADDRSTRLEN];
```

Comment:
Similarly, please use a macro for INET_ADDRSTRLEN and INET_ADDRSTRLEN

Reply:
Ack